Doc/library/asyncio-task.rst

```diff
@@ -518,6 +518,30 @@ Running Tasks Concurrently
       and there is no running event loop.
 
 
+Eager Task Factory
+==================
+
+.. function:: eager_task_factory(loop, coro, *, name=None, context=None)
+
+    A task factory for eager task execution.
+
+    When using this factory (via ``loop.set_task_factory(asyncio.eager_task_factory)``),
+    coroutines that are able to complete synchronously (without suspending)
+    are returned immediately as a completed :class:`Future`.
+
+    A regular :class:`Task` is returned otherwise, at the first suspension of *coro*.
```

Comment:
This probably needs more explanation. I'd explain that it always tries to execute the coroutine immediately, until it either blocks, returns, or raises, and that when it blocks it returns a Task, when it returns/raises it returns a Future.

Returning a Future is still something I am skeptical about, since the task factory is global (for everything using the same event loop); I'll have to re-read the Issue. I'll come back here for the rest of the code review after I've had time for that.